at_lookup/lib/src/at_lookup_impl.dart

```diff
@@ -275,12 +274,20 @@ class AtLookupImpl implements AtLookUp {
       // Setting the errorCode and errorDescription to default values.
       var errorCode = 'AT0014';
       var errorDescription = 'Unknown server error';
-      if (verbResult.contains('-')) {
-        if (verbResult.split('-')[0].isNotEmpty) {
-          errorCode = verbResult.split('-')[0];
-        }
-        if (verbResult.split('-')[1].isNotEmpty) {
-          errorDescription = verbResult.split('-')[1];
+      try {
+        var errorMap = jsonDecode(verbResult);
+        errorCode = errorMap['errorCode'];
+        errorDescription = errorMap['errorDescription'];
+      } on FormatException {
+        // Catching the FormatException to preserve backward compatibility - responses without jsonEncoding.
+        // TODO: Can we remove the below catch block in next release once all the servers are migrated to new version.
```

Comment:
We agreed that the decoding parameter should be supported in the non-JSON format as well

Reply:
@gkc: With reference to the following [PR-841](https://github.com/atsign-foundation/at_server/pull/841), After merging the PR-841 changes, the error messages from the server will be JSON encoded. Hence marked a TODO because server will not send the non json error responses.